File: WebFormCases2/Utils/SqlHelper.cs
using System;
using System.Collections.Generic;
using System.Configuration;
using System.Data;
using System.Data.SqlClient;
using System.Linq;
using System.Text;


namespace MyWebFormCases.Utils

{
	public static class SqlHelper
	{
		private static string constr = ConfigurationManager.ConnectionStrings["EntityDb"].ConnectionString;

		public static int ExcuteNunQuery(string sql,params SqlParameter[] sqlParameters) {
			using (SqlConnection con=new SqlConnection(constr))
			{
				using (SqlCommand com=new SqlCommand(sql,con))
				{
					com.Parameters.AddRange(sqlParameters);
					con.Open();
					return com.ExecuteNonQuery();
				}
			}
		}

		public static int ExcuteNunQueryPro(string sql, params SqlParameter[] sqlParameters)
		{
			using (SqlConnection con = new SqlConnection(constr))
			{
				using (SqlCommand com = new SqlCommand(sql, con))
				{
					com.CommandType = CommandType.StoredProcedure;
					com.Parameters.AddRange(sqlParameters);
					con.Open();
					return com.ExecuteNonQuery();
				}
			}
		}

		public static DataTable getDataTable(string sql, params SqlParameter[] sqlParameters) {

			using (SqlDataAdapter adapter = new SqlDataAdapter(sql, constr)) {

				adapter.SelectCommand.Parameters.AddRange(sqlParameters);
				DataTable table = new DataTable();
				adapter.Fill(table);
				return table;
			}
		}
		public static DataTable getDataTablePro(string sql, params SqlParameter[] sqlParameters)
		{

			using (SqlDataAdapter adapter = new SqlDataAdapter(sql, constr))
			{

				adapter.SelectCommand.Parameters.AddRange(sqlParameters);
				adapter.SelectCommand.CommandType = CommandType.StoredProcedure;
				DataTable table = new DataTable();
				adapter.Fill(table);
				return table;
			}
		}


		public static Object ExcuteScalar(string sql, params SqlParameter[] sqlParameters) {
			using (SqlConnection con = new SqlConnection(constr)) {
				using (SqlCommand com = new SqlCommand(sql, con)) {
					com.Parameters.AddRange(sqlParameters);
					con.Open();
					return com.ExecuteScalar();
				}

			}

		}
		public static Object ExcuteScalarPro(string sql, params SqlParameter[] sqlParameters)
		{
			using (SqlConnection con = new SqlConnection(constr))
			{
				using (SqlCommand com = new SqlCommand(sql, con))
				{
					com.CommandType = CommandType.StoredProcedure;
					com.Parameters.AddRange(sqlParameters);
					con.Open();
					return com.ExecuteScalar();
				}

			}

		}

		public static SqlDataReader GetSqlDataReader(string sql, params SqlParameter[] sqlParameters) {

		
			SqlConnection con = new SqlConnection(constr);
			using (SqlCommand com = new SqlCommand(sql, con)) {
				try
				{
					con.Open();
			
				
					
					return com.ExecuteReader(System.Data.CommandBehavior.CloseConnection);
				}
				catch (Exception)
				{
					con.Close();
					con.Dispose();

					throw;
				}	

			}

		}
		public static SqlDataReader GetSqlDataReaderPro(string sql, params SqlParameter[] sqlParameters)
		{

		
			SqlConnection con = new SqlConnection(constr);
			using (SqlCommand com = new SqlCommand(sql, con))
			{
				try
				{
					com.CommandType = CommandType.StoredProcedure;
					con.Open();

					com.Parameters.AddRange(sqlParameters);
				
					return com.ExecuteReader(System.Data.CommandBehavior.CloseConnection);
				}
				catch (Exception)
				{
					con.Close();
					con.Dispose();

					throw;
				}

			}

		}
	}
}

File: VbCases/utils/SqlHelper.vb
Imports System.Data.SqlClient

Module SqlHelper
    Private constr As String = ConfigurationManager.ConnectionStrings("supplyModel").ConnectionString

    Function ExcuteNunQuery(ByVal sql As String, ParamArray sqlParameters As SqlParameter()) As Integer
        Using con As SqlConnection = New SqlConnection(constr)

            Using com As SqlCommand = New SqlCommand(sql, con)
                com.Parameters.AddRange(sqlParameters)
                con.Open()
                Return com.ExecuteNonQuery()
            End Using
        End Using
    End Function

    Function ExcuteNunQueryPro(ByVal sql As String, ParamArray sqlParameters As SqlParameter()) As Integer
        Using con As SqlConnection = New SqlConnection(constr)

            Using com As SqlCommand = New SqlCommand(sql, con)
                com.CommandType = CommandType.StoredProcedure
                com.Parameters.AddRange(sqlParameters)
                con.Open()
                Return com.ExecuteNonQuery()
            End Using
        End Using
    End Function

    Function getDataTable(ByVal sql As String, ParamArray sqlParameters As SqlParameter()) As DataTable
        Using adapter As SqlDataAdapter = New SqlDataAdapter(sql, constr)
            adapter.SelectCommand.Parameters.AddRange(sqlParameters)
            Dim table As DataTable = New DataTable()
            adapter.Fill(table)

        End Using
#Disable Warning BC42105 ' Function doesn't return a value on all code paths
    End Function
#Enable Warning BC42105 ' Function doesn't return a value on all code paths

    Function getDataTablePro(ByVal sql As String, ParamArray sqlParameters As SqlParameter()) As DataTable
        Using adapter As SqlDataAdapter = New SqlDataAdapter(sql, constr)
            adapter.SelectCommand.Parameters.AddRange(sqlParameters)
            adapter.SelectCommand.CommandType = CommandType.StoredProcedure
            Dim table As DataTable = New DataTable()
            adapter.Fill(table)
            Return table
        End Using
    End Function

    Function ExcuteScalar(ByVal sql As String, ParamArray sqlParameters As SqlParameter()) As Object
        Using con As SqlConnection = New SqlConnection(constr)

            Using com As SqlCommand = New SqlCommand(sql, con)
                com.Parameters.AddRange(sqlParameters)
                con.Open()
                Return com.ExecuteScalar()
            End Using
        End Using
    End Function

    Function ExcuteScalarPro(ByVal sql As String, ParamArray sqlParameters As SqlParameter()) As Object
        Using con As SqlConnection = New SqlConnection(constr)

            Using com As SqlCommand = New SqlCommand(sql, con)
                com.CommandType = CommandType.StoredProcedure
                com.Parameters.AddRange(sqlParameters)
                con.Open()
                Return com.ExecuteScalar()
            End Using
        End Using
    End Function

    Function GetSqlDataReader(ByVal sql As String, ParamArray sqlParameters As SqlParameter()) As SqlDataReader
        Dim con As SqlConnection = New SqlConnection(constr)

        Using com As SqlCommand = New SqlCommand(sql, con)

            Try
                con.Open()
                Return com.ExecuteReader(System.Data.CommandBehavior.CloseConnection)
            Catch __unusedException1__ As Exception
                con.Close()
                con.Dispose()
                Throw
            End Try
        End Using
    End Function

    Function GetSqlDataReaderPro(ByVal sql As String, ParamArray sqlParameters As SqlParameter()) As SqlDataReader
        Dim con As SqlConnection = New SqlConnection(constr)

        Using com As SqlCommand = New SqlCommand(sql, con)

            Try
                com.CommandType = CommandType.StoredProcedure
                con.Open()
                com.Parameters.AddRange(sqlParameters)
                Return com.ExecuteReader(System.Data.CommandBehavior.CloseConnection)
            Catch __unusedException1__ As Exception
                con.Close()
                con.Dispose()
                Throw
            End Try
        End Using
    End Function
End Module

